ingesters/HttpIngester/hec.go

```diff
@@ -169,13 +170,23 @@ loop:
 			return
 		}
 	}
-	if ackRequested(r) {
-		json.NewEncoder(w).Encode(ack{
-			ID: strconv.FormatUint(atomic.AddUint64(&hh.id, 1), 10),
-		})
+	if doAck, ch := ackRequested(r); doAck {
+		hh.writeAck(ch, w)
 	}
 }
 
+func (hh *hecHandler) writeAck(channel string, w http.ResponseWriter) {
+	hh.ackLock.Lock()
+	defer hh.ackLock.Unlock()
+	if _, ok := hh.ackIds[channel]; !ok {
+		hh.ackIds[channel] = 0
+	}
+	hh.ackIds[channel] = hh.ackIds[channel] + 1
+	json.NewEncoder(w).Encode(ack{
```

Comment:
I think we need to release the acklock before writing to the network.  A slow client would block all other clients